File: need4spd/euler_41.py
def isPrime(n):
    n = abs(int(n))
    if n < 2:
        return False
    if n == 2: 
        return True  
    if not n & 1: 
        return False
    for x in range(3, int(n**0.5)+1, 2):
        if n % x == 0:
            return False
    return True

# euler 24
import itertools
 
l=list(range(1,10))
permutations=list(itertools.permutations(l,9))
 
p=list()
for t in permutations:
    s="".join(str(n) for n in t)
    p.append(s)

p.sort()

print (p[0])
result = 0

for n in p:

  if isPrime(n):
    print(n)
    if result < n:
      result = n

print (result)

File: need4spd/euler_41.pl
use List::Permutor;

use strict;
use warnings;
 
sub isPrime
{
    my ($number) = @_;
    $number = abs $number;
     
    my $sqrt = sqrt $number;
    my $d = 2;
    while (1)
    {
        return 0 if ( $number % $d == 0 );
        return 1 if ( $d > $sqrt );
        $d++;
    }
}

our $result = 0;

sub getPrime {
    my ($number) = @_;
 
    while (my @p = $number->next) {
    my $str = join "", @p;
    if (isPrime($str)) {
        if($result < $str) {
            $result = $str;
        }
    }
}   
}

my $perm = new List::Permutor qw /1 2 3 4/;
&getPrime($perm);
$perm = new List::Permutor qw /1 2 3 4 5/;
&getPrime($perm);
$perm = new List::Permutor qw /1 2 3 4 5 6/;
&getPrime($perm);
$perm = new List::Permutor qw /1 2 3 4 5 6 7/;
&getPrime($perm);
$perm = new List::Permutor qw /1 2 3 4 5 6 7 8/;
&getPrime($perm);
$perm = new List::Permutor qw /1 2 3 4 5 6 7 8 9/;
&getPrime($perm);


print "$result";